core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -192,6 +197,56 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
+    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+
+    /**
+     * Transfers the text, prefix, regex, analyzer, fieldname, size and shard size settings from the
+     * original {@link SuggestionBuilder} to the target {@link SuggestionContext}
+     */
+    public void populateCommonFields(MapperService mapperService,
```

Comment:
Yes, might even become private by moving some of the common stuff from the build methods here like in @abeyad suggested.